### ISSUE TITLE
GraphQL schema name collision caused by autogenerated `Order` input parameter

### ISSUE THREAD

**Issue by ThunbergOlle** [parse-community/parse-server#8802]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description
When autogenerating the GraphQL schema, Parse creates input types for each class. All the generated input types has a trailing `Input` word. Except for the `Order` input type. Meaning that if you have a class called `Item` the sorting input type will be called `ItemOrder` 

This becomes problematic since it is common, when for example, building e-commerce sites, to have collections ending with "Order" (as in buy-order). ItemOrder, GiftCardOrder etc. 

Because of this, if you have two classes called `Item` and `ItemOrder` it will cause a name collision in the schema. The result of this being that `ItemOrder` (the collection) defaults its type to `Object` causing an error when trying to access its collection properties through the GraphQL API. 

Here is an example of how creating a class called `Item` reserving the type name `ItemOrder`
```graphql
items(
where: ItemWhereInput
order: [ItemOrder!]
skip: Int
after: String
first: Int
before: String
last: Int
options: ReadOptionsInput
): ItemConnection!
```

If we also have a `ItemOrder` collection, it logs this warning:
```
warn: Type ItemOrder could not be added to the auto schema because it collided with an existing type.
```

And the type for the `ItemOrder` (collection) query is:
```graphql
itemOrder(
id: ID!
options: ReadOptionsInput
): Object
```

### Steps to reproduce
1. Create a new parse project with GraphQL
3. Create a class called Item
4. Create a class called ItemOrder

### Solutions
A potential solution is to follow the naming conventions parse already uses for the other input types. Meaning that we can add `Input` to the end.

Example:
```graphql
items(
where: ItemWhereInput
order: [ItemOrderInput!]
skip: Int
after: String
first: Int
before: String
last: Int
options: ReadOptionsInput
): ItemConnection!
```

It should be noted that this is a breaking change if the input type is being used as follows:
```graphql
query myQuery($order: [ItemOrder!]){
  items(order: $order){
    count
  }
}
```

This would have to be changed to this: 
```graphql
query myQuery($order: [itemOrderInput!]){
  items(order: $order){
    count
  }
}
```

**Comment by Moumouls** [comment]
Hi @ThunbergOlle, interesting! What was your solution here? Your suggestion of using "input" isn't quite right, as "Input" has a specific meaning in GraphQL. I believe ending with "Enum" would be more appropriate.

The main concern here could be a significant breaking change if developers explicitly use the "Order" type in GQL queries or mutations.

BTW @ThunbergOlle there is a feature that allow if i remember to adjust the name of the class exposed in the GQL API

**Comment by ThunbergOlle** [comment]
Hello! @Moumouls Thanks for the response. 

I agree with you that enum is more appropriate for the ordering. 👍 

It's true that this would be a breaking change and therefore perhaps not doable. Feel free to close the issue if that is the case 😄 

Yes, there is a feature by attaching classConfigs. We've utilised this for some plural cases which the autogenerated names were unable to handle. For instance:
```js
      {
        className: 'ProductData',
        query: {
          findAlias: 'productDatas',
        },
      },
```

It's true that we could probably use this for XXOrder as well, but we strive to keep database collections and graphql namings as close as possible. And therefore we feel like making an abstraction / renaming the core "Order"-tables creates unnecessary complexity. 🥲
